### PR TITLE
General sanity in function `seriesOutliersDetectTukey`

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 414e6eb6002cff04e40e77dbdd39de593d58e1ff with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60535/414e6eb6002cff04e40e77dbdd39de593d58e1ff/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by rschu1ze** [comment]
@bhavnajindal FYI: `seriesOutliersDetectTukey` was changed from percent (0 - 100) to percentile (0.00 - 1.00).

File: src/Functions/seriesOutliersDetectTukey.cpp [hunk 1]
```diff
@@ -73,30 +70,26 @@ class FunctionSeriesOutliersDetectTukey : public IFunction
...
-            min_percentile = p_min / 100;
+            min_percentile = arguments[1].column->getFloat64(0);
+            if (isnan(min_percentile) || !isFinite(min_percentile) || min_percentile < min_percentile_lower_bound|| min_percentile > max_percentile_upper_bound)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "The second argument of function {} must be in range [2.0, 98.0]", getName());
 
-            Float64 p_max = arguments[2].column->getFloat64(0);
-            if (isnan(p_max) || !isFinite(p_max) || p_max < min_quartile || p_max > max_quartile || p_max < min_percentile * 100)
+            max_percentile = arguments[2].column->getFloat64(0);
+            if (isnan(max_percentile) || !isFinite(max_percentile) || max_percentile < min_percentile_lower_bound || max_percentile > max_percentile_upper_bound || max_percentile < min_percentile)
                 throw Exception(ErrorCodes::BAD_ARGUMENTS, "The third argument of function {} must be in range [2.0, 98.0]", getName());
```
Comment (antaljanosbenjamin) [inline src/Functions/seriesOutliersDetectTukey.cpp:88]: The exception message still mentions `2.0` and `98.0`.

File: src/Functions/seriesOutliersDetectTukey.cpp [hunk 2]
```diff
@@ -73,30 +70,26 @@ class FunctionSeriesOutliersDetectTukey : public IFunction
...
 
-            max_percentile = p_max / 100;
-
-            auto k_val = arguments[3].column->getFloat64(0);
-            if (k_val < 0.0 || isnan(k_val) || !isFinite(k_val))
+            k = arguments[3].column->getFloat64(0);
+            if (k < 0.0 || isnan(k) || !isFinite(k))
                 throw Exception(ErrorCodes::BAD_ARGUMENTS, "The fourth argument of function {} must be a positive number", getName());
```
Comment (antaljanosbenjamin) [inline src/Functions/seriesOutliersDetectTukey.cpp:92]: In the docs it says `non-negative`, but here it says positive. The check allows `0`, so either change the error message; or change the docs and the check to prohibit `0` as `k`.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

**Comment by antaljanosbenjamin** [comment]
Also, because we basically changed the range of possible input parameters, I think it make sense to include it in the changelog.

File: src/Functions/seriesOutliersDetectTukey.cpp (hunk 1 above)
Comment (rschu1ze) [reply src/Functions/seriesOutliersDetectTukey.cpp:88]: Thanks, fixed.

**Comment by antaljanosbenjamin** [comment]
> Agree in general, but v24.2 is the first release with seriesOutliersDetectTukey so there are no backwards compat concerns.

Makes sense for public, but private we have an existing release with this function. I am not sure if the private changelog is available or not for users, but for use it can be useful.

**Pull request merged**